packages/x-adapter/README.md

```diff
@@ -1,13 +1,467 @@
+
+To create an `EndpointAdapter` you can use the `endpointAdapterFactory` function. This function will
+receive an `EndpointAdapterOptions` object containing all the needed data to perform and map a
+request, and return a function that when invoked will trigger the request. The options that can be
+configured are:
+
+- `endpoint`: The URL that the `httpClient` uses. It can be either a string or a mapper function
+  that dynamically generates the URL string using the request data.
+- `httpClient`: A function that will receive the endpoint and request options such as the parameters
+  and will perform the request, returning a promise with the unprocessed response data.
+- `defaultRequestOptions`: Default values for the endpoint configuration. You can use it to define
+  if a request is cancelable, a unique id to identify it, anything but the `endpoint` can be set.
+  Check
+  [EndpointAdapterOptions](https://github.com/empathyco/x/blob/main/packages/x-adapter/src/endpoint-adapter/types.ts)
+  to see the available options.
+- `requestMapper`: A function to transform the unprocessed request into parameters the API can
+  understand.
+- `responseMapper`: A function to transform the API response into data that your project can
+  understand.
+
+<br>
+
+#### Basic adapter implementation
+
+In this example we have a simple request mapper that will add a `q` parameter to the endpoint's url
+to perform the request. If you check the function call below, you will see the query parameter
+passed.
+
+###### Types definition
+
+```ts
+// API data models
+interface ApiRequest {
+  q?: string;
+  id?: number;
+}
+interface ApiSearchResponse {
+  products: ApiProduct[];
+  total: number;
+}
+interface ApiProduct {
+  id: number;
+  title: string;
+  price: number;
+}
+
+// App's data models
+interface AppSearchRequest {
+  query: string;
+}
+interface AppSearchResponse {
+  products: AppProduct[];
+  total: number;
+}
+interface AppProduct {
+  id: string;
+  name: string;
+  price: number;
+}
+```
+
+###### Adapter's factory function implementation
+
+```ts
+import { endpointAdapterFactory } from '@empathyco/x-adapter';
+
+export const searchProducts = endpointAdapterFactory({
+  endpoint: 'https://dummyjson.com/products/search',
+  requestMapper({ query }: Readonly<AppSearchRequest>): ApiRequest {
+    return {
+      q: query // the request will be triggered as https://dummyjson.com/products/search?q=phone
+    };
+  },
+  responseMapper({ products, total }: Readonly<ApiSearchResponse>): AppSearchResponse {
+    return {
+      products: products.map(product => {
+        return {
+          id: product.id.toString(),
+          name: product.title,
+          price: product.price
+        };
+      }),
+      total: total
+    };
+  }
+});
+
+// Function call
+async function searchOnClick() {
+  const response = await searchProducts({ query: 'phone' });
+  this.items = response.products;
+}
+```
+
+<br>
+
+#### Using a dynamic endpoint
```

Comment:
RE-posting because I believe this comment was lost when modifying the file 😅
I would probably invert this section. Start with the simple use case and then go to the advanced one.

Using a dynamic endpoint
If you need to generate an endpoint url dynamically, you can add parameters inside curly brackets to the endpoint string. By default these parameters will be replaced using the request data. If a parameter is not found inside the request, an empty string will be used.
```ts
export const getItemById = endpointAdapterFactory({
  endpoint: 'https://dummyjson.com/{section}/{id}',
  // ... rest of options to configure
  }
});
getItemById({ section: "products", id: 1 }); // 'https://dummyjson.com/products/1'
getItemById({ section: "quotes", id: 3 }); // 'https://dummyjson.com/quotes/3'
getItemById({ section: "quotes" }); // 'https://dummyjson.com/quotes/'
For more complex use cases, you can use a mapper function. This function receives the request, and must return the URL string.

export const getProductById = endpointAdapterFactory({
  endpoint: ({ id }: GetProductByIdRequest) => 'https://dummyjson.com/products/' + id,
  // ... rest of options to configure
});
```